packages/mrml-python/src/lib.rs

```diff
@@ -10,19 +10,19 @@ use mrml::prelude::parser::noop_loader::NoopIncludeLoader;
 use pyo3::exceptions::PyIOError;
 use pyo3::prelude::*;
 
-#[pyclass]
+#[pyclass(frozen)]
 #[derive(Clone, Debug, Default)]
 pub struct NoopIncludeLoaderOptions;
 
-#[pyclass]
+#[pyclass(frozen)]
 #[derive(Clone, Debug, Default)]
 pub struct MemoryIncludeLoaderOptions(HashMap<String, String>);
 
-#[pyclass]
+#[pyclass(frozen)]
```

Comment:
I've added (frozen) to all the classes to make them immutable. This is the recommended way to keep code thread safe, and may become the default in the future: https://pyo3.rs/main/class.html?highlight=frozen#frozen-classes-opting-out-of-interior-mutability

Reply:
Nice, thanks

Reply:
I know there are quite a few changes in this PR. Let me know if any of this is interesting or if you'd like me to break it up a bit. 

I think there are technically some API breaks in here, but since the type definition on RenderOptions wasn't working, I suspect most people are just using the main API.